scripts/Muon/maxent_model.py

```diff
@@ -1,70 +1,122 @@
 from __future__ import (absolute_import, division, print_function)
+
 from six import iteritems
+
 import mantid.simpleapi as mantid
 
 
-class MaxEntModel(object):
-    # A simple class to hold the MaxEnt algorithm
+class MaxEntWrapper(object):
 
-    def __init__(self):
-        self.alg=mantid.AlgorithmManager.create("MuonMaxent")
-        self.alg.initialize()
-        self.alg.setChild(True)
+    """
+    A class to wrap the different parts
+    of the MaxEnt and its preprocessing.
+    This keeps the main MaxEnt class simple.
+    """
+
+    def __init__(self, maxent):
+        self.name = "MaxEnt"
+        self.model = maxent
 
-    def setInputs(self,inputs,runName):
-        self.inputs=inputs
-        for name,value in iteritems(self.inputs):
-            self.alg.setProperty(name,value)
-        self.run=runName
+    def loadData(self, inputs):
+        """
+        store the data in the wrapper for later
+        """
+        if "phaseTable" in inputs:
+            self.phaseTable = inputs["phaseTable"]
+        else:
+            self.phaseTable = None
+        if "maxent" in inputs:
+            self.maxent = inputs["maxent"]
+        else:
+            self.maxent = None
+        self.model.setRun(inputs["Run"])
 
     def execute(self):
-        self.alg.execute()
+        """
+        runs the relevant parts of the MaxEnt and the preprocessing
+        """
+        if self.phaseTable is not None:
+            self.model.makePhaseTable(self.phaseTable)
+
+        if self.maxent is not None:
+            self.model.MaxEntAlg(self.maxent)
 
     def cancel(self):
-        self.alg.cancel()
+        self.model.cancel()
 
     def output(self):
-        self.addOutput("OutputWorkspace")
-        self.addOutput("OutputPhaseTable")
-        self.addOutput("OutputDeadTimeTable")
-        self.addOutput("ReconstructedSpectra")
-        self.addOutput("PhaseConvergenceTable")
-
-    def addOutput(self,name):
-        if name in self.inputs:
-            mantid.AnalysisDataService.addOrReplace( self.inputs[name],self.alg.getProperty(name).value)
-        else:
-            return
-        if mantid.AnalysisDataService.doesExist(self.run):
-            group=mantid.AnalysisDataService.retrieve(self.run)
-        else:
-            mantid.GroupWorkspaces(OutputWorkspace=self.run)
+        return
```

Comment:
Is this supposed to be an empty return ?

Reply:
yep